### PR TITLE
Download code + EDA

### PR DESCRIPTION
**Pull request by ben-heil**
Sorry that there's a couple things going on here; the design decisions in the downloading code were influenced by the EDA so I kind of just picked a cutoff point

This PR adds code to download citation data from COCI, MeSH heading information from Entrez, build networks from the citation, and performs EDA on those networks

Changes:
- Add gitignore file
- Add download_data.sh to download MeSH metadata
- Add environment.yml
- Start visualizing networks in network_eda (wow I like plotly but the file sizes get so big so fast)
- Add build_network.py to combine MeSH headings and citation data into citation network objects
- Add download_article_metadata.py to download MeSH info
- Add download_citations.sh to download the COCI citation data

### CODE REVIEW

File: download_data.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+BIO_SUBHEADINGS=("botany" "cell biology" "computational biology" "cryobiology" "developmental biology" "ecology" "exobiology" "genetics" "laboratory animal science" "microbiology" "natural history" "neurobiology" "parasitology" "photobiology" "radiobiology" "sociobiology" "synthetic biology" "zoology")
+INFORMATICS=("cheminformatics" "computational biology" "consumer health informatics" "dental informatics" "medical informatics" "nursing informatics" "public health informatics")
+ALGORITHMS=("artificial intelligence" "latent class analysis" "cellular automata")
```
Comment (jjc2718) [inline download_data.sh:3]: How did you choose these terms?

File: src/build_network.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    print('Building graphs...')
+    # For reasons that are unclear to me, this is faster than either inserting all
+    # edges at once or constructing the graph in one shot
+    for file_path in tqdm(glob.glob(f'{args.data_dir}/*')):
+        citation_list = pd.read_csv(file_path)
+        for heading in headings:
+            heading_dois = heading_to_dois[heading]
+            for citing, cited in zip(citation_list['citing'], citation_list['cited']):
+                if citing in heading_dois and cited in heading_dois:
+                    heading_to_graph[heading].add_edge(citing, cited)
```
Comment (jjc2718) [inline src/build_network.py:92]: Does this mean you're only counting citations within headings (i.e. from a comp bio paper to another comp bio paper, but not from a comp bio paper to an ML paper)? I wonder how many citations aren't under the same heading, or if there's a way to count that.

File: src/download_article_metadata.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                        help='The MeSH term whose article info should be downloaded')
+    parser.add_argument('--overwrite',
+                        help="Set this flag to redownload terms that "
+                             "already have data",
+                        action='store_true')
+
+    args = parser.parse_args()
+
+    # Make the term safe for putting in a path
+    path_term = '_'.join(args.mesh_term.split(' '))
```
Comment (jjc2718) [inline src/download_article_metadata.py]: Could you just do:
```suggestion
    path_term = args.mesh_term.replace(' ', '_')
```
Or could there be multiple spaces between parts of a term?

**Review by jjc2718 — APPROVED** [review]
Looks good! This seems like it will be an interesting project.

Would it be possible to convert the notebook to a Python script? That would make it easier to review in the future, since I can't load the diff for a file with >1M lines. (I can still load/view the notebook itself in nbviewer but I can't make inline comments that way)

Small comments about the notebook:
* Do you want to make a comment somewhere about the articles with 0 papers cited? Or are you going to investigate that more in the future?
* Did you see any qualitative difference between the citation distributions in the different fields/headings? They all look roughly the same to me, but I haven't looked super closely.

File: download_data.sh (hunk 1 above)
Comment (ben-heil) [reply download_data.sh:3]: I picked them while browsing the MeSH tree view. I'm currently working on more principled ways of picking headings

File: src/build_network.py (hunk 2 above)
Comment (ben-heil) [reply src/build_network.py:92]: Right, my guess is that ~2/3 of the citations are to papers outside the field, but I plan on running a full analysis on that in the future

File: src/download_article_metadata.py (hunk 3 above)
Comment (ben-heil) [reply src/download_article_metadata.py]: Yeah, that would definitely make more sense

**Pull request merged**